contracts/voting/dao-voting-native-staked/src/contract.rs

```diff
@@ -360,9 +564,128 @@ pub fn query_list_stakers(
     to_binary(&ListStakersResponse { stakers })
 }
 
+pub fn query_is_active(deps: Deps) -> StdResult<Binary> {
+    let threshold = ACTIVE_THRESHOLD.may_load(deps.storage)?;
+    if let Some(threshold) = threshold {
+        let denom = DENOM.load(deps.storage)?;
+        let actual_power = STAKED_TOTAL.may_load(deps.storage)?.unwrap_or_default();
+        match threshold {
+            ActiveThreshold::AbsoluteCount { count } => to_binary(&IsActiveResponse {
+                active: actual_power >= count,
+            }),
+            ActiveThreshold::Percentage { percent } => {
+                // percent is bounded between [0, 100]. decimal
+                // represents percents in u128 terms as p *
+                // 10^15. this bounds percent between [0, 10^17].
+                //
+                // total_potential_power is bounded between [0, 2^128]
+                // as it tracks the balances of a cw20 token which has
+                // a max supply of 2^128.
+                //
+                // with our precision factor being 10^9:
+                //
+                // total_power <= 2^128 * 10^9 <= 2^256
+                //
+                // so we're good to put that in a u256.
+                //
+                // multiply_ratio promotes to a u512 under the hood,
+                // so it won't overflow, multiplying by a percent less
+                // than 100 is gonna make something the same size or
+                // smaller, applied + 10^9 <= 2^128 * 10^9 + 10^9 <=
+                // 2^256, so the top of the round won't overflow, and
+                // rounding is rounding down, so the whole thing can
+                // be safely unwrapped at the end of the day thank you
+                // for coming to my ted talk.
+                let total_potential_power: cosmwasm_std::SupplyResponse =
+                    deps.querier
+                        .query(&cosmwasm_std::QueryRequest::Bank(BankQuery::Supply {
+                            denom,
+                        }))?;
+                let total_power = total_potential_power
+                    .amount
+                    .amount
```

Comment:
I know this is coming from cosmwasm std but it's always funny to see `.amount.amount` lol

Reply:
I know. SO ANNOYING.